tests/test_tlsfuzzer_expect.py

```diff
@@ -1285,52 +1285,140 @@ def test_process_with_tls13_unallowed_extension(self):
+        c_ks = key_share_gen(GroupName.secp256r1)
+        ext = ClientKeyShareExtension().create([c_ks])
+        client_hello.extensions.append(ext)
+        ext = SupportedVersionsExtension().create([(3, 3), (3, 4)])
+        client_hello.extensions.append(ext)
         state.handshake_messages.append(client_hello)
         state.msg_sock = mock.MagicMock()
 
         rndbuf=bytearray(32)
         rndbuf[-8:] = TLS_1_2_DOWNGRADE_SENTINEL
+        ext = []
+        ext.append(SrvSupportedVersionsExtension().create((3, 4)))
 
-        msg = ServerHello().create(version=(3, 4),
+        msg = ServerHello().create(version=(3, 3),
                                    random=rndbuf,
                                    session_id=bytearray(0),
-                                   cipher_suite=4,
-                                   extensions=None)
+                                   cipher_suite=
+                                   CipherSuite.TLS_AES_128_GCM_SHA256,
+                                   extensions=ext)
 
         self.assertTrue(exp.is_match(msg))
 
-        with self.assertRaises(AssertionError):
+        with self.assertRaises(AssertionError) as e:
             exp.process(state, msg)
 
+        self.assertIn("downgrade protection sentinel but shouldn",
+                      str(e.exception))
+
+    def test_process_with_default_and_tls_1_3_with_1_1_downgrade_sentinel(self):
+        # verify that if server sends TLS 1.3 server hello, we check that it
+        # does not include downgrade sentinel
+        exp = ExpectServerHello()
+
+        state = ConnectionState()
+        client_hello = ClientHello()
+        client_hello.extensions = []
+        client_hello.cipher_suites = [CipherSuite.TLS_AES_128_GCM_SHA256]
+        ext = SupportedGroupsExtension().create([GroupName.secp256r1])
+        client_hello.extensions.append(ext)
+        c_ks = key_share_gen(GroupName.secp256r1)
+        ext = ClientKeyShareExtension().create([c_ks])
+        client_hello.extensions.append(ext)
+        ext = SupportedVersionsExtension().create([(3, 3), (3, 4)])
+        client_hello.extensions.append(ext)
+        state.handshake_messages.append(client_hello)
+        state.msg_sock = mock.MagicMock()
+
+        rndbuf=bytearray(32)
+        rndbuf[-8:] = TLS_1_1_DOWNGRADE_SENTINEL
+        ext = []
+        ext.append(SrvSupportedVersionsExtension().create((3, 4)))
+
+        msg = ServerHello().create(version=(3, 3),
+                                   random=rndbuf,
+                                   session_id=bytearray(0),
+                                   cipher_suite=
+                                   CipherSuite.TLS_AES_128_GCM_SHA256,
+                                   extensions=ext)
+
+        self.assertTrue(exp.is_match(msg))
+
+        with self.assertRaises(AssertionError) as e:
+            exp.process(state, msg)
+
+        self.assertIn("downgrade protection sentinel but shouldn",
```

Comment:
clarify message ?
also shouldn -> shouldn't ?

Reply:
it's just check that the raised exception is the one we expect, not just any AssertionError, but yeah the `'t` could be added - will do

Reply:
the exception and message in question:
https://github.com/tomato42/tlsfuzzer/blob/c28cdd361f3255f1a76d652c79e0af04c4541103/tlsfuzzer/expect.py#L663-L665